packages/@aws-cdk/aws-apprunner/lib/observability-configuration.ts

```diff
@@ -0,0 +1,131 @@
+import * as cdk from '@aws-cdk/core';
+import { Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnObservabilityConfiguration } from './apprunner.generated';
+
+
+/**
+ * The implementation provider for tracing App Runner services.
+ */
+export enum TracingVendor {
+  /**
+   * No Tracing Vendor
+   */
+  NONE = 'NONE',
+
+  /**
+   * AWS X-Ray
+   */
+  AWSXRAY = 'AWSXRAY'
+}
+
+/**
+ * Properties of the AppRunner Observability Configuration
+ */
+export interface ObservabilityConfigurationProps {
+  /**
+   * The implementation provider for tracing App Runner services.
+   *
+   * @default Vendor.None
+   */
+  readonly traceConfiguration?: TracingVendor;
+
+  /**
+    * The name for the Observability Configuration.
+    *
+    * @default - a name generated by CloudFormation
+    */
+  readonly observabilityConfigurationName?: string;
+}
+
+/**
+ * Represents the AppRunner Observability Configuration
+ */
+export interface IObservabilityConfiguration extends cdk.IResource {
+  /**
+   * The ARN of the Observability Configuration.
+   * @attribute
+   */
+  readonly observabilityConfigurationArn: string;
+
+  /**
+   * The Name of the Observability Configuration.
+   * @attribute
+   */
+  readonly observabilityConfigurationName: string;
+}
+
+/**
+ * The App Runner Observability Configuration
+ *
+ * @resource AWS::AppRunner::ObservabilityConfiguration
+ */
+export class ObservabilityConfiguration extends cdk.Resource implements IObservabilityConfiguration {
+  /**
+   * Import an observability configuration by ARN.
+   */
+  public static fromObservabilityConfigurationArn(scope: Construct, id: string, observabilityConfigurationArn: string): IObservabilityConfiguration {
+    class Import extends cdk.Resource implements IObservabilityConfiguration {
+      public readonly observabilityConfigurationArn = observabilityConfigurationArn;
+      public readonly observabilityConfigurationName = observabilityConfigurationArn.split('/')[1];
```

Comment:
This ARN could potentially be a token so this won't be handled correctly. We may need to use a Lazy function to produce this later.